### PR TITLE
feat: add `Gamma` distribution

### PR DESCRIPTION
**Pull request by ikrommyd**
Adds `Gamma` distribution
We choose the same parametrization [with ROOT](https://root.cern/doc/master/classRooGamma.html)
Related to #512 

## Proposed Changes

- Implement  Gamma distribution

## Tests added

- Modify `tests/serialize/test_basics.py`  and add `tests/test_pdf_pdf.py`

## Checklist

- [ ] change approved
- [x] implementation finished
- [x]  docs updated
- [x] correct namespace imported
- [x] tests added
- [x] CHANGELOG updated
- [x] (if new public method/class) docs in rst file updated?
- [x] (if large change) tutorial added/updated?

### CODE REVIEW

File: zfit/models/dist_tfp.py [hunk 1]
```diff
@@ -933,3 +933,105 @@ class BifurGaussPDFRepr(BasePDFRepr):
...
+        obs: ztyping.ObsTypeInput,
+        *,
+        extended: ExtendedInputType = None,
+        norm: NormInputType = None,
+        name: str = "Gamma",
+    ):
+        """Gamma distribution.
+
+        The gamma shape is parametrized here with `gamma`, `beta` and `mu`.
+        We follow the same parametrization `as RooFit <https://root.cern.ch/doc/master/classRooGamma.html>`_.
```
Comment (jonas-eschle) [inline zfit/models/dist_tfp.py:955]:
```suggestion
        The gamma shape is parametrized here with `gamma`, `beta` and `mu`, following
        the same parametrization `as RooFit <https://root.cern.ch/doc/master/classRooGamma.html>`_.
```

File: zfit/models/dist_tfp.py [hunk 2]
```diff
@@ -933,3 +933,105 @@ class BifurGaussPDFRepr(BasePDFRepr):
...
+
+            f(x \\mid \\gamma, \\beta, \\mu) = (x - \\mu)^{\\gamma - 1} \\exp{\\left(-\\frac{x - \\mu}{\\beta}\\right)} / Z
+
+        with the normalization over [0, inf] of
+
+        .. math::
+
+            Z = \\Gamma(\\gamma) \\beta^{\\gamma}
+
+        The normalization changes for different normalization ranges
```
Comment (jonas-eschle) [inline zfit/models/dist_tfp.py:968]:
```suggestion
        The normalization changes for different normalization ranges and `Z=1` for the unnormalized shape. 
```

File: zfit/models/dist_tfp.py [hunk 3]
```diff
@@ -933,3 +933,105 @@ class BifurGaussPDFRepr(BasePDFRepr):
...
+            dist_params=dist_params,
+            obs=obs,
+            params=params,
+            name=name,
+            extended=extended,
+            norm=norm,
+        )
+
+    @property
+    def distribution(self):
```
Comment (jonas-eschle) [inline zfit/models/dist_tfp.py:1016]: I would suggest to refactor this and remove the custom `distribution` property: above, `distribution` can be any callable that takes the parameters and returns a distribution, so we can just define this above in the init, approx
```
def distribution(mu, sigma,...):
    return TransformedDist(......)
```

(reduces duplicated code and streamlines the pdf)

**Review by jonas-eschle — CHANGES_REQUESTED** [review]
Good overall, one refactoring and minor docs

File: zfit/models/dist_tfp.py (hunk 3 above)
Comment (ikrommyd) [reply zfit/models/dist_tfp.py]: Please take a look and let me know if this is what you were aiming for

**Review by jonas-eschle — APPROVED** [review]
Perfect, that's what I had in mind, looks good! LGTM

**Pull request merged**